Sources/SwiftParser/Types.swift

```diff
@@ -194,15 +194,21 @@ extension Parser {
       return RawTypeSyntax(RawMissingTypeSyntax(arena: self.arena))
     }
 
-    // '.Type', '.Protocol', '?', '!', and '[]' still leave us with type-simple.
     var loopCondition = LoopProgressCondition()
     while loopCondition.evaluate(currentToken) {
-      if !stopAtFirstPeriod, let (period, type) = self.consume(
-        if: { [.period, .prefixPeriod].contains($0.tokenKind) },
-        followedBy: { $0.isContextualKeyword(["Type", "Protocol"])}
-      ) {
-        base = RawTypeSyntax(RawMetatypeTypeSyntax(
-          baseType: base, period: period, typeOrProtocol: type, arena: self.arena))
+      if !stopAtFirstPeriod, self.at(.period) || self.at(.prefixPeriod) {
```

Comment:
This allows `A.Type.B` (which is currently not parsed). And the structure would be like `(memberType (metatypeType (typeIdentifier)))`. I'm totally OK with it, but we should at least have a test case?

Reply:
Sure, I was pretty lazy with non-metatype test additions (ie. I didn't add any). Do we want a diagnostic for the `A.Type.B` case, ie. any member type with a metatype suggesting to add backpacks?

Reply:
@rintaro We are working toward modeling and diagnosing stuff with `A.Type.B` in Sema (apple/swift#62775, apple/swift#62319). I am going to add tests for these to the Swift parser in #1171.

Reply:
Thanks @AnthonyLatsis 🙇. Sorry, I didn't realize you were making changes for this in the Swift parser as well. Sounds like we'll be able to have ASTGen just create the naive AST and then have Sema diagnose.